src/app/teacher/teacher-run-list-item/teacher-run-list-item.component.ts

```diff
@@ -44,18 +49,35 @@ export class TeacherRunListItemComponent implements OnInit {
         this.run.highlighted = false;
       }, 7000);
     }
+    this.subscribeToTagUpdated();
   }
 
-  ngAfterViewInit() {
+  subscribeToTagUpdated(): void {
+    this.subscriptions.add(
+      this.projectTagService.tagUpdated$.subscribe((tagThatChanged: Tag) => {
+        const tagOnProject = this.run.project.tags.find((tag: Tag) => tag.id === tagThatChanged.id);
+        if (tagOnProject != null) {
```

Comment:
When is this null check necessary?

Reply:
If the run/project does not have the tag that was updated, the tagOnProject will be null.